### ISSUE TITLE
onStart() event fires when live was not streaming

### ISSUE THREAD

**Issue by featheast** [flowplayer/flash#227]
Hi there,
We are using Flowplayer for live streaming using RTMP which works for quite a long time so far. After upgrading our lib from 3.2.16 -> 3.2.18, one of the things we have noticed that changed the behaviour is the onStart() event.

Previously, it will not get fired until the streaming start to play, which will only happen when someone is actually broadcasting. Now, however, this event will always fire regardless anyone broadcasting or not. We have used this event to determine if the streaming is on/off, and provide relevant feedback based on this, but it is not working any more.

Can you please check if there is any change of the code that causing this event or there is another way that we can achieve the same purpose?

Thanks,
Y.L.

**Comment by danrossi** [comment]
Yes something was added for red5 servers to trigger a start event as it doesn't provide metadata. The server is in fact sending a play start event. So I guess check if there is a time value added ?

**Comment by featheast** [comment]
Hi Dan,

Yes, I had a look at issue #28 and ny looking at the code, onStart event will be triggered after 1.5s if no metadata is received.
But back to the event itself, is it supposed to be the meaning of onStart is for the player start to play?
Or as the documentation describes "This fires at the point at which playback commences."

I don't quite agree that the onStart event should be triggered just for the purpose of lacking metadata.

Can you please elaborate how I can check if a time value is added to the event? I'm using javascript.

Or I guess I can switch back my RTMP version back to 3.2.14.

Thanks
Y.L.

**Comment by danrossi** [comment]
on start is triggered when metadata is available, with some live streams none is available so it will dispatch after 1500ms to display the video screen. because the server is sending a play event this event will therefore be dispatched.

**Comment by danrossi** [comment]
you can check out the sources to change it if you want or 

onStart: function() {
if (this.getTime() > 0) {

}

}

try that
